### PR TITLE
Split up Linux Daemon disable and stop

### PR DESCRIPTION
**Pull request by Hoverbear**
In some situations the user may have the systemd service enabled, but the service stopped. This should workaround that.

Was part of #57 but uhhh, it's not a related change.

### CODE REVIEW

File: src/action/base/configure_nix_daemon_service.rs [hunk 1]
```diff
@@ -207,21 +207,52 @@ impl Action for ConfigureNixDaemonService {
...
-                    Command::new("systemctl")
-                        .args(["disable", SERVICE_SRC, "--now"])
-                        .stdin(std::process::Stdio::null()),
-                )
-                .await
-                .map_err(|e| ConfigureNixDaemonServiceError::Command(e).boxed())?;
+                let socket_is_active = is_active("nix-daemon.socket").await?;
+                let socket_is_enabled = is_enabled("nix-daemon.socket").await?;
+                let service_is_active = is_active("nix-daemon.service").await?;
+                let service_is_enabled = is_enabled("nix-daemon.service").await?;
```
Comment (Hoverbear) [inline src/action/base/configure_nix_daemon_service.rs:215]: I guess with this I could join the stop/disable again....

**Review by cole-h — APPROVED** [review]
Seems fine. One minor comment that may or may not be pertinent.

File: src/action/base/configure_nix_daemon_service.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Hoverbear) [reply src/action/base/configure_nix_daemon_service.rs:215]: Ohhhhhhhhh.....

[the remark this answers is not in the document]
Comment (cole-h) [reply src/action/base/configure_nix_daemon_service.rs:215]: And unfortunately, `linked` makes it return `1`, so matching on the "is return code errored" won't (easily) work here either.

**Review by cole-h — APPROVED** [review]
Probably fine for now. Though if people report having other outputs, it might make more sense to match on what it looks like when it's not enabled?

**Comment by Hoverbear** [comment]
I agree!

**Pull request merged**